numpy/lib/_shape_base_impl.pyi

```diff
@@ -57,6 +65,19 @@ class _SupportsArrayWrap(Protocol):
     @property
     def __array_wrap__(self) -> _ArrayWrap: ...
 
+# Protocol for array-like objects that preserve their type through split operations.
+# Requires shape (with __getitem__ for axis access), ndim for dimensional checks
+# in hsplit/vsplit/dsplit, swapaxes for axis manipulation, and __getitem__ for
+# slicing. Examples: pandas DataFrame, xarray DataArray.
+@type_check_only
+class _SupportsSplitOps(Protocol):
+    @property
+    def shape(self) -> tuple[int, ...]: ...
+    @property
+    def ndim(self) -> int: ...
+    def swapaxes(self, axis1: SupportsIndex, axis2: SupportsIndex, /) -> Self: ...
```

Comment:
I'm not sure if this would cause any issues for pandas, but the `axis{1,2}` are in contravariant (input) positions, so it would be more general to annotate them as `: int`.